iterating-over-data/iterating-over-dictionaries.md

```diff
@@ -0,0 +1,200 @@
+    "beverage": "Fancy Lemonade"
+}
+
+for dish, food in menu.items():
+    print(f"The special {dish} for tonight is the {food}.")
+```
+
+```python
+menu = {
+  "Brussels Sprouts": 18.99,
+  "Fancy Lemonade": 56.00
+}
+
+for food, price in menu.items():
+    taxed_price = price * 1.101
+    print(f"The cost of {food} is {taxed_price}")
+
+print("That sure was a meal!")
+```
+
+## More Examples
+
+Our loop logic can be more complex, and include conditionals, calculations, and function calls.
+
+```python
+menu = {
+  "Brussels Sprouts": 18.99,
+  "Fancy Lemonade": 56.00,
+  "Summer Salad": 48.50,
+  "Tomato Soup": 18.50
+}
+
+for food, price in menu.items():
+    taxed_price = price * 1.101
+    if taxed_price > 20:
+        print(f"{food} costs more than $20.")
+    else:
+        print(f"{food} costs less than $20")
+
+print("That sure was a meal!")
+```
+
+Loops will often live in functions. We can use a `for` loop on a list that was passed into a function, too!
+
+```python
+def consider_specials(specials):
+    for dish, special in specials.items():
+        print(f"For the special {dish}, why not consider the {special} tonight?")
+
+menu = {
+    "appetizer": "Brussels Sprouts",
+    "beverage": "Fancy Lemonade"
+}
+
+consider_specials(menu)
+```
+
+## `for` Loops With Only Keys Doesn't Need `.items()`
+
+When we need to iterate over a dictionary and we **don't** need a reference to the value for each key-value pair, we can use this alternative syntax:
+
+```python
+for my_key in my_dict:
+    print(my_key)
+```
+
+This syntax only names a local variable `my_key`, which will be each key, and does not require us to call `.items()` on our dictionary.
+
+We can see in this example, our code does not raise a `ValueError` when we use this syntax! Of course, we can access the 
+
+```python
+menu = {
+    "appetizer": "Brussels Sprouts",
+    "beverage": "Fancy Lemonade"
+}
+
+for dish in menu:
+    print(f"The special {dish} for tonight is the {menu[dish]}.")
+```
+
+Callout: If this syntax accomplishes the same thing, which `for` loop style should we prefer? It is always up to the developer, but most Python developers prefer having keys and values available and iterating with `.items()`.
+
+## Iterating With `.keys()` or `.values()`
+
+Dictionaries have a method `.keys()`, which returns a list of every key in the dictionary.
+
+If we wanted to iterate over all the keys in a dictionary, we could call `.keys()` on it, and then iterate over it as an array.
+
+Consider this example:
+
+```python
+menu = {
+    "appetizer": "Brussels Sprouts",
+    "beverage": "Fancy Lemonade"
+}
+
+print("What are the categories on the menu tonight?")
+categories = []
+
+for category in menu.keys():
+    categories.append(category)
+
+print(categories)
+```
+
+Callout: For this example, alternatively, we could have printed the categories inside the for loop. We thought it would be nice to see this pattern again, as it accomplishes something slightly different.
```

Comment:
```suggestion
### !callout-info
## Loop Patterns
For this example, alternatively, we could have printed the categories inside the for loop. We thought it would be nice to see this pattern again, as it accomplishes something slightly different.
### !end-callout
```